packages/vite/src/node/cli.ts

```diff
@@ -218,13 +220,7 @@ cli
           'serve',
           'development'
         )
-        await preview(
-          config,
-          cleanOptions(options) as {
-            host?: string
-            port?: number
-          }
-        )
+        await preview(config, cleanOptions(options))
```

Comment:
Could you jump into `preview` function and replace there the `{ host?: string; port?: number }` with `Pick<ServerOptions, 'host' | 'port'>`

https://github.com/iheyunfei/vite/blob/a1537b7b8be8bec7ad0f926f87bb18f50ff169bd/packages/vite/src/node/preview.ts#L21

And you need to import `ServerOptions` from `'.'`

=> `import { ResolvedConfig, ServerOptions } from '.'` in line 6

Reply:
@Shinigami92 It looks like ServerOptions['host'] `host?: string | boolean` is not assignable for arguments of `preview`, whose `host` type is `host?: string;` .

Reply:
So do we have a type incompatibility here? And until now it wasn't noticed due to the `as`.
Or do we need to adjust something else?

Reply:
Yes, you are right. Another type problem should be fixed here.

Reply:
Fix it too. Thanks for all the suggestions.